client/src/AppBundle/Controller/ManageController.php

```diff
@@ -11,6 +16,30 @@
  */
 class ManageController extends AbstractController
 {
+    /** @var array<ServiceAvailabilityAbstract> */
+    private $services = [];
+
+    public function __construct(
+        ContainerInterface $container,
```

Comment:
Will it be problematic injecting the container in v4/5? At the least, it's won't be best practice - perhaps we can inject the `env` parameter directly?

Reply:
In v4/5 we'll be able to inject `ParameterBagInterface` which lets us get application parameters in a clean dependency-injecty way. But unfortunately that's not supported in v3 so I thought using the container for now would give us the easiest upgrade path.

Reply:
Happy with that